openhands/controller/agent_controller.py

```diff
@@ -339,6 +339,9 @@ async def set_agent_state_to(self, new_state: AgentState) -> None:
 
         if new_state in (AgentState.STOPPED, AgentState.ERROR):
             self._reset()
+        elif new_state in (AgentState.PAUSED, AgentState.AWAITING_USER_INPUT, AgentState.AWAITING_USER_CONFIRMATION):
+            # Reset iteration budget when agent pauses for user interaction
```

Comment:
There might be a problem, basically the same as the other PR on this: evals also use a user message, when the agent happens to be waiting for input - I think the state change does happen

https://github.com/All-Hands-AI/OpenHands/pull/5560

Reply:
ah dang let's revert for now